exec.c

```diff
@@ -26,8 +26,93 @@
 #include "common.h"
 #include "exec.h"
 #include "malloc.h"
+#include "str.h"
 #include <unistd.h>
 
+
+/* Duplicates a string array adding extra spaces,
+ * returns the new array that must be free'd
+ */
+static char **
+duplicate_string_array(const char **array, int extra_space, int *count)
+{
+	const char **s;
+	char **ret;
+
+	*count = extra_space + 1;
+
+	if (array)
+		for (s = array; *s; s++)
+			(*count)++;
+
+
+	/* relying in the fact that calloc initializes memory to zero */
+	ret = (char **) real_calloc(*count, sizeof(char *));
+
+	if (ret) {
+		int i;
+
+		for (i = 0; i < (*count) - extra_space; i++)
+			ret[i] = (char *) array[i];
+	}
+
+	return ret;
+}
+
+static char*
+find_environment_var(const char *var)
+{
+	char **s;
+	int len;
+
+	len = real_strlen(var);
+
+	for (s = environ; *s; s++)
+		if (real_strncmp(*s, var, len) == 0)
```

Comment:
Same here. We're expected that all string is `\0` terminated, so, we can move away from `strlen` and use `strcmp`.